tests/test_backend.py

```diff
@@ -281,3 +290,26 @@ def test_qpu_backend(sequence):
 
     results = remote_results.results
     assert results[0].sampling_dist == {"00": 1.0}
+
+
+def test_qpu_back_with_context_manager(sequence):
+    connection = _MockConnection()
+    unique_test_id = "unique-id-string"
+
+    with pytest.warns(DeprecationWarning, match="From v0.18"):
+        seq = sequence.switch_device(replace(DigitalAnalogDevice, max_runs=10))
+
+    with QPUBackend(seq, connection).open_submission(unique_test_id) as qpub:
+        remote_results = qpub.run(job_params=[{"runs": 10}])
+        assert isinstance(remote_results, RemoteResults)
+        # confirms context manager assigns ID
+        assert qpub.submission_id == unique_test_id
```

Comment:
nit: you could assert the submission_id is reset when you exit the context manager